### PR TITLE
Update to Astro 3.4

### PR DESCRIPTION
**Pull request by ryannelsonsmartcontract**
## Closing issues
Closes [FRONT-4353](https://smartcontract-it.atlassian.net/browse/FRONT-4353)

## Description
Updates Astro to `3.4.0` (currently on `2.10.15`). This requires our deployment server to be running at least Node `18.14.1`.

## Considerations
Not using ViewTransitions. I want to, but there is a bug where navigating using ViewTransitions unloads styles from some client components (Algovera Search, Table of Contents). There has been [documentation on this issue](https://github.com/withastro/astro/issues/8361) and a [supposed fix](https://github.com/withastro/astro/pull/8776), but updating and trying different configurations hasn't worked for me.

### CODE REVIEW

File: astro.config.ts [hunk 1]
```diff
@@ -10,12 +11,19 @@ import sitemap from "@astrojs/sitemap"
 // https://astro.build/config
 export default defineConfig({
   site: "https://docs.chain.link",
-  // trailingSlash: 'never',
-  integrations: [preact(), react(), sitemap({ changefreq: "daily" }), mdx()],
+  integrations: [
+    preact({
+      include: ["**/preact/*"],
```
Comment (Alejandro-Larumbe) [inline astro.config.ts:16]: Question: isn't this indicating that we are having the preact components in a preact folder? Since the components are not currently organized this way, is this doing anything currently?

File: src/components/Quickstart/StickyHeaderObserver.tsx [hunk 2]
```diff
@@ -95,7 +95,7 @@ const StickyHeaderObserver: React.FC = () => {
     }
   }, [$navBarInfo.height])
 
-  return <div ref={thisRef} className={styles.stickyBar} hidden={true} data-sticky />
+  return <div ref={thisRef} className={styles.stickyBar} hidden={true} data-sticky></div>
```
Comment (Alejandro-Larumbe) [inline src/components/Quickstart/StickyHeaderObserver.tsx:98]: Is this change necessary?

File: src/features/feeds/components/pause-notice/PauseNotice.tsx [hunk 3]
```diff
@@ -55,7 +55,7 @@ export const PauseNotice = ({
       return (
         <>
           <div class={styles.banner + " " + styles[type]}>
-            <img class={styles.icon} src={icons[type]}></img>
+            <img class={styles.icon} src={icons[type].src}></img>
```
Comment (Alejandro-Larumbe) [inline src/features/feeds/components/pause-notice/PauseNotice.tsx:58]:
```suggestion
            <img class={styles.icon} src={icons[type].src}/>
```

File: src/features/feeds/components/pause-notice/PauseNotice.tsx [hunk 4]
```diff
@@ -71,7 +71,7 @@ export const PauseNotice = ({
             class={styles.banner + " " + styles.tooltip + " " + styles[type]}
             tooltip-text="This feed is paused due to lack of attestation data."
           >
-            <img class={styles.iconSmall} src={icons[type]}></img>
+            <img class={styles.iconSmall} src={icons[type].src}></img>
```
Comment (Alejandro-Larumbe) [inline src/features/feeds/components/pause-notice/PauseNotice.tsx:74]: Ditto

**Review by Alejandro-Larumbe — APPROVED** [review]

File: src/components/Quickstart/StickyHeaderObserver.tsx (hunk 2 above)
Comment (ryannelsonsmartcontract) [reply src/components/Quickstart/StickyHeaderObserver.tsx:98]: Semantic change, just don't think divs should be self closing

File: astro.config.ts (hunk 1 above)
Comment (ryannelsonsmartcontract) [reply astro.config.ts:16]: Specifying `include` is necessary for the integrations to work properly on 3.0, actually having them in the folders is optional

File: src/components/Quickstart/StickyHeaderObserver.tsx (hunk 2 above)
Comment (ryannelsonsmartcontract) [reply src/components/Quickstart/StickyHeaderObserver.tsx:98]: [P.S. this will be deleted in this PR anyways](https://github.com/smartcontractkit/documentation/pull/1584/files#diff-8e0d9c121aa3be61e192bb5dc7ff49965a968c54cb42463af7b6ef360a13344d)

**Comment by ryannelsonsmartcontract** [comment]
Found a bug with global style files not being included on build specifically, working on fixing now

File: astro.config.ts [hunk 5]
```diff
@@ -24,9 +32,8 @@ export default defineConfig({
       ],
       // Wrap tables in div with overflow supported
       [rehypeWrapAll, { selector: "table", wrapper: "div.overflow-wrapper" }],
-    ],
+    ] as any[],
```
[the remark this answers is not in the document]
Comment (ryannelsonsmartcontract) [reply astro.config.ts:35]: This was a weird one, it seemed to be an issue with Astro's own types post-3.0

**Comment by ryannelsonsmartcontract** [comment]
Having trouble using ViewTransition because the AI search styling breaks for some unknown reason

**Review by dwightjl — APPROVED** [review]
